src/hooks/useFlipCard.ts

```diff
@@ -4,14 +4,14 @@ interface UseFlipCardProps {
   frontDeps: Array<React.ComponentState>;
   backDeps: Array<React.ComponentState>;
 }
+
 export default function useFlipCard({ frontDeps, backDeps }: UseFlipCardProps) {
   const [isFrontSide, setFrontSide] = useState(true);
 
   const flipCard = () => {
     setFrontSide((prev) => !prev);
   };
 
-  // TODO: JSON.stringify 써도 되는지 물어보기
```

Comment:
배열(의 요소)에 변화가 있는지를 파악하기 위해 `JSON.stringify(array)`를 의존성 배열에 넣는 경험이 있습니다. 렛서가 넣어주신 값도 배열이라, 의도한 대로 동작할 것으로 보여요.

그런데 useEffect로 관리하기보다, cvc input에서
- onFocus -> setBackSide(true)
- onBlur -> setBackSide(false)
이렇게 관리하면 어떨까 싶습니다.

언제 effect를 사용할지, 언제 event handling에 대한 좋은 문서가 있어 공유드립니다.

⭐️추천⭐️: https://react.dev/learn/you-might-not-need-an-effect
When you’re not sure whether some code should be in an Effect or in an event handler, ask yourself why this code needs to run. Use Effects only for code that should run because the component was displayed to the user. In this example, the notification should appear because the user pressed the button, not because the page was displayed! Delete the Effect and put the shared logic into a function called from both event handlers